src/PostgREST/ApiRequest.hs

```diff
@@ -191,11 +191,19 @@ userApiRequest schema req reqBody =
     where lastWord = last $ T.split (=='.') key
 
   headerRange = if singular && method == "GET" then singletonRange 0 else rangeRequested hdrs
-  urlOffsetRange = rangeGeq . fromMaybe (0::Integer) $
-    readBSMaybe =<< join (lookup "offset" qParams)
-  urlRange = restrictRange
-    (readBSMaybe =<< join (lookup "limit" qParams))
-    urlOffsetRange
+  replaceLast x s = T.intercalate "." $ L.init (T.split (=='.') s) ++ [x]
```

Comment:
Actually I think we're safe here either way because split always returns a list with at least one item, for instance `split (=='.') ""` returns `[""]`. In this case replaceLast will just concatenate a dot and another value to the original string.